sys/auto_init/auto_init.c

```diff
@@ -93,6 +98,10 @@
 
 void auto_init(void)
 {
+#ifdef MODULE_TLSF_MALLOC
+    /* NOTE: this should be initialized before any other modules */
```

Comment:
This is right, but is not enough. It should be initialized before the C library itself (that is what `.preinit_array` is for).